xapian-core/backends/honey/honey_database.cc

```diff
@@ -205,19 +205,16 @@ HoneyDatabase::get_doclength_upper_bound() const
 Xapian::termcount
 HoneyDatabase::get_wdf_upper_bound(const string& term) const
 {
-    // We don't store per-term wdf upper bounds currently, only a per-database
-    // wdf bound.  However, the collection frequency of the term provides a
-    // second upper bound (since collection frequency is the sum of the wdf and
-    // wdf >= 0), so pick the tighter of these bounds.
     Xapian::termcount wdf_bound = version_file.get_wdf_upper_bound();
-    // It's unlikely wdf is always 0, but when it is there's no need to check
-    // the collection frequency.
+    // It's unlikely wdf is always 0, but when it is there's no need to do any
+    // further work.
     if (usual(wdf_bound != 0)) {
-	Xapian::termcount coll_freq;
-	get_freqs(term, NULL, &coll_freq);
-	if (coll_freq < wdf_bound) {
-	    wdf_bound = coll_freq;
-	}
+	// We don't store per-term wdf upper bounds currently, but
+	// HoneyPostListTable can provide an upper bound based on termfreq,
+	// coll_freq, and the first wdf value, which more often than not is
+	// actually the exact bound (in 77% of cases in an example database of
+	// wikipedia data).
+	wdf_bound = min(wdf_bound, postlist_table.get_wdf_upper_bound(term));
```

Comment:
You've seem to have a commit from master on your branch too, which means those changes are appearing in the diff making it harder to review, and also mean that github gets a conflict when it tries to merge master with the branch to CI test it.

I'm not sure how you managed to get that commit there, but whatever you did wasn't quite right.

I think in this case, it's simplest to fix this by rewriting the history of the branch to drop this extra commit.  Generally you shouldn't rewrite history of a branch under review because it makes it harder to follow what's changed, but in this case it seems the least bad answer.

You should be able to rebase your branch interactively onto master like so:

```
git rebase -i master
```

That'll open an editor with the list of rebased commits - I'd expect the extra commit to automatically drop out as it's present on current master already, but if not just delete the line for it and save the file.  Exit the editor and git will rewrite the branch for you.

You can need to force push it to the PR so the remote will accept the rewritten history (to re-iterate, this isn't something you should usually be doing):

```
git push --force
```